app/services/subscriptions/dates_service.rb

```diff
@@ -45,9 +45,9 @@ def to_datetime
       return @to_datetime if @to_datetime
 
       @to_datetime = customer_timezone_shift(compute_to_date, end_of_day: true)
-      terminated_at = subscription.terminated_at
+      terminated_at = subscription.terminated_at&.change(usec: 0)
 
-      if subscription.terminated? && @to_datetime > terminated_at && billing_at >= terminated_at
+      if subscription.terminated? && @to_datetime > terminated_at && billing_at&.change(usec: 0) >= terminated_at
```

Comment:
Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.